hudi-hadoop-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -473,7 +481,10 @@ private <T extends Comparable<T>> HoodieColumnRangeMetadata<T> getColumnRangeInF
         .reduce(HoodieColumnRangeMetadata::merge).get();
   }
 
-  private static Comparable<?> convertToNativeJavaType(PrimitiveType primitiveType, Comparable<?> val) {
+  private static Comparable<?> convertToNativeJavaType(PrimitiveType primitiveType, Comparable<?> val, ValueMetadata valueMetadata) {
+    if (valueMetadata.getValueType() != ValueType.V1) {
+      return valueMetadata.standardizeJavaTypeAndPromote(val);
+    }
```

Comment:
Could `ParquetUtils#readColumnStatsFromMetadata` avoid checking the value type or index version and let the caller to do the conversion, i.e., reading parquet metadata and returning the raw stats should not care about the Hudi-related semantics?